tools/ExtensionsMetadataGenerator/test/ExtensionsMetadataGeneratorTests/ExtensionsMetadataGeneratorTests.cs

```diff
@@ -27,21 +27,37 @@ public void Generator_DifferentTargetFrameworks()
+            Console.SetOut(log);
 
-            string extensionsJson = File.ReadAllText(outputFile);
-            JObject json = JObject.Parse(extensionsJson);
-
-            // We expect to see the Startups from this assembly (running on Core 2.0),
-            // plus the two from the 2.1 and 2.2 test projects.
-            JToken extensions = json["extensions"];
-            int startups = extensions.Count();
-            Assert.Equal(4, startups);
-
-            Assert.Single(extensions, e => e["name"].ToString() == "Foo" && e["typeName"].ToString().StartsWith("ExtensionsMetadataGeneratorTests.FooWebJobsStartup, ExtensionsMetadataGeneratorTests"));
-            Assert.Single(extensions, e => e["name"].ToString() == "BarExtension" && e["typeName"].ToString().StartsWith("ExtensionsMetadataGeneratorTests.BarWebJobsStartup, ExtensionsMetadataGeneratorTests"));
-            Assert.Single(extensions, e => e["name"].ToString() == "Startup" && e["typeName"].ToString().StartsWith("TestProject_Core21.Startup"));
-            Assert.Single(extensions, e => e["name"].ToString() == "Startup" && e["typeName"].ToString().StartsWith("TestProject_Core22.Startup"));
+            try
+            {
+                Program.Main(new[] { sourcePath, outputFile });
+
+                string extensionsJson = File.ReadAllText(outputFile);
+                JObject json = JObject.Parse(extensionsJson);
+
+                // We expect to see the Startups from this assembly (running on Core 2.0),
+                // plus the two from the 2.1 and 2.2 test projects.
+                JToken extensions = json["extensions"];
+                int startups = extensions.Count();
+                Assert.Equal(4, startups);
+
+                Assert.Single(extensions, e => e["name"].ToString() == "Foo" && e["typeName"].ToString().StartsWith("ExtensionsMetadataGeneratorTests.FooWebJobsStartup, ExtensionsMetadataGeneratorTests"));
+                Assert.Single(extensions, e => e["name"].ToString() == "BarExtension" && e["typeName"].ToString().StartsWith("ExtensionsMetadataGeneratorTests.BarWebJobsStartup, ExtensionsMetadataGeneratorTests"));
+                Assert.Single(extensions, e => e["name"].ToString() == "Startup" && e["typeName"].ToString().StartsWith("TestProject_Core21.Startup"));
+                Assert.Single(extensions, e => e["name"].ToString() == "Startup" && e["typeName"].ToString().StartsWith("TestProject_Core22.Startup"));
+
+                // We cannot read TestProject_Razor.dll because it has a dependency we cannot find. Make sure
+                // we log correctly even though we've skipped this assembly.
+                Assert.Contains("Cannot load 'Microsoft.AspNetCore.Razor.Runtime'. Aborting assembly resolution.", log.ToString());
+                Assert.Contains("Could not evaluate 'TestProject_Razor.dll' for extension metadata. If this assembly contains a Functions extension, ensure that all dependent assemblies exist in ", log.ToString());
```

Comment:
Once I added the Razor project, this test ran into the same StackOverflowException that customers were reporting. Now I make sure that we succeed and log the problem correctly.